pkg/netflow/flowaggregator/aggregator_test.go

```diff
@@ -127,10 +127,22 @@ func TestAggregator(t *testing.T) {
 	// Test aggregator Stop
 	assert.False(t, expectStartExisted)
 	aggregator.Stop()
-	time.Sleep(500 * time.Millisecond)
-	stoppedMu.Lock()
-	assert.True(t, expectStartExisted)
-	stoppedMu.Unlock()
+
+	waitStopTimeout := time.After(2 * time.Second)
+	waitStopTick := time.Tick(100 * time.Millisecond)
+stopLoop:
+	for {
+		select {
+		case <-waitStopTimeout:
+			assert.Fail(t, "timeout waiting for aggregator to be stopped")
+		case <-waitStopTick:
+			stoppedMu.Lock()
+			if expectStartExisted {
+				break stopLoop
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
if this breaks `stopLoop` without finishing it's loop, you need to call `stoppedMu.Unlock()` before breaking

Reply:
Good catch, indeed not great to break before releasing the mutex.

Note, in this case, we are at the end of the test, so, the lock is not causing issue since the mutex is going to GCed anyway, but agree, it's not great to break without releasing the lock.

Fixed here https://github.com/DataDog/datadog-agent/pull/10976/commits/fd42df8552b156e1c1ce9c90a86d8678684d118e